### PR TITLE
Improve dev deployments on infra clusters

### PR DESCRIPTION
**Pull request by kovayur**
## Description
This PR contains various changes related to development builds and deployments on infra clusters.

**The main change is**: instead of manually setting the fleet manager image tag built on GHA to make it available on an infra cluster, we build it locally and push it to the internal openshift image registry. Image push results into an image stream tag to be available on the cluster.

As a result, only the following steps are required to deploy fleet* components on an infra cluster:
1. Provision a cluster
2. `make deploy/bootstrap`
3. `make deploy/dev` 

Miscellaneous changes:
1. Remove fleet manager image build on GHA (as a consequence of the main change)
1. Drop gke support (I don't see the reason of supporting non-Openshift infra clusters)
1. Delete image push script (not used for a long time)

## Checklist (Definition of Done)

- [ ] Unit and integration tests added
- [ ] Added test description under `Test manual`
- [ ] Documentation added if necessary (i.e. changes to dev setup, test execution, ...)
- [ ] CI and all relevant tests are passing
- [ ] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`
- [ ] Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.
- [ ] Add secret to app-interface Vault or Secrets Manager if necessary
- [ ] RDS changes were e2e tested [manually](../docs/development/howto-e2e-test-rds.md)
- [ ] Check AWS limits are reasonable for changes provisioning new resources
- [ ] (If applicable) Changes to the dp-terraform Helm values have been reflected in the addon on integration environment

## Test manual

**TODO:** Add manual testing efforts

```
# To run tests locally run:
make db/teardown db/setup db/migrate
make ocm/setup
make verify lint binary test test/integration
```

### CODE REVIEW

File: dev/env/defaults/02-infra.env [hunk 1]
```diff
@@ -5,9 +5,18 @@ is_infra_cluster() {
     server=$(try_kubectl config view --minify -o jsonpath='{.clusters[0].cluster.server}')
     if [[ "$server" =~ infra\.rox\.systems ]]; then
         return 0
-    else
-        return 1
     fi
+    if [[ "$server" =~ openshiftapps\.com || "$server" =~ devshift\.org ]]; then
+        # Selected cluster is ROSA or OSD. Need to make sure that it's an infra cluster.
```
Comment (ludydoo) [inline dev/env/defaults/02-infra.env:10]: Does this work for Openshift on GCP ?

Comment (kovayur) [reply dev/env/defaults/02-infra.env:10]: Yes, with some caveats*. OCP case is handled a few lines above. ROSA and OSD clusters have different hosts, so I added them here.

\* I can't `docker login` to external registry on "Openshift 4.x" cluster without adding a CA certificate to the docker daemon even if I set "Should trusted certificates be created" to true. I stated this in the readme file.

Comment (kovayur) [reply dev/env/defaults/02-infra.env:10]: Perhaps it's colima specific. The trust store is missing the CA certificate used in infra, I haven't dug very deep.

Comment (kovayur) [reply dev/env/defaults/02-infra.env:10]: Docker registry error is fixed in stackrox/automation-flavors#264

**Review by ludydoo — APPROVED** [review]

File: dev/env/defaults/02-infra.env (hunk 1 above)
Comment (kovayur) [reply dev/env/defaults/02-infra.env:10]: I have solved the above issue on infra and modified the readme file accordingly.

**Pull request merged**